### ISSUE TITLE
Conflict with methods (mongoose-delete ?)

### ISSUE THREAD

**Issue by www-chique** [francescov1/mongoose-tsgen#45]
Hi there. I'm back with another problem.
The tool works amazing out of the box, and most of the other issues you've already helped me in the past and ironed them out.
But it came to this one:

I am using [mongoose-delete](https://www.npmjs.com/package/mongoose-delete) plugin.
And on the models where I've used the plugin, it shows a Typescript error on the generated interface such as follows:
**Image 1**
![image](https://user-images.githubusercontent.com/29447216/113581388-b4313b80-9644-11eb-9bcc-e45c15e5554a.png)

**Image 2**
It also generated the following type, which is correct
![image](https://user-images.githubusercontent.com/29447216/113581454-c9a66580-9644-11eb-8151-42f1d514e3e3.png)

Now I am a newbie at this, so I am not entirely sure what would be the correct solution.
But if I do this, it gets fixed. (Copy pasted the line from Image 2 into Image 1)
![image](https://user-images.githubusercontent.com/29447216/113581671-15f1a580-9645-11eb-9987-aabfe170c02d.png)

You're extremely good at this, so I'll look up-to your solution to this problem or how you would approach this.

Many thanks, and appreciate this tool so much every day.

**Comment by francescov1** [comment]
Hey @www-chique, good to hear from you again, glad to hear everything else is going well with the tool!

Currently in a bit of a time crunch with my other work but I'm looking to tackle this early next week! This plugin is actually something I've been meaning to setup for one of my projects so your question comes at a perfect time. I'll keep you posted!

**Comment by www-chique** [comment]
> 
> 
> Hey @www-chique, good to hear from you again, glad to hear everything else is going well with the tool!
> 
> Currently in a bit of a time crunch with my other work but I'm looking to tackle this early next week! This plugin is actually something I've been meaning to setup for one of my projects so your question comes at a perfect time. I'll keep you posted!

Sure, there is absolutely no rush. For these kind of special cases, I always run a custom fix afterwards as a workaround.
I just let you know, so maybe the tool gets more robust with time.
Thanks a lot again.

**Comment by francescov1** [comment]
I investigated your issue, it occurs because `mongoose.Document` defines the default document delete function, which is different from the generated version. To get around this, I'm thinking we could do something like this:

```typescript
// before
export interface EmailTemplateDocument extends mongoose.Document<mongoose.Types.ObjectId>, EmailTemplateMethods { ... }

// after
export interface EmailTemplateDocument extends Omit<mongoose.Document<mongoose.Types.ObjectId>, keyof EmailTemplateMethods>, EmailTemplateMethods { ... }
```

Could you confirm that the fix above fixes your issue? It essentially removes any properties from `mongoose.Document` that are already defined in `EmailTemplateMethods`. 

I'd like to look into any potential consequences & edge cases with implementing this. Should be able to do that in the next few days, once that's confirmed it will be a quick fix. One thing to mention is that we recently released mongoose-tsgen v8.0.0. This introduced a few (very minor) breaking changes, but requires mongoose 5.11.19 or higher. I suggest you upgrade to this version in the meantime so that you can get the fix easily once its released, since we won't be updating older versions of mongoose-tsgen moving forward.

**Comment by www-chique** [comment]
Hey. I can confirm, this fixes the problem. I tried it, works great.
Thank you very much.

**Comment by francescov1** [comment]
Fixed in 8.3.0 🚀 Ended up finding a simpler option, while improving general typing!
